test/common/stats/thread_local_store_test.cc

```diff
@@ -1149,6 +1149,30 @@ TEST_F(StatsMatcherTLSTest, RejectPrefixNoDot) {
   EXPECT_MEMORY_LE(mem_consumed, 3500000);
 }
 
+TEST_F(StatsMatcherTLSTest, DoNotRejectHiddenRegex) {
+  stats_config_.mutable_stats_matcher()->mutable_exclusion_list()->add_patterns()->MergeFrom(
```

Comment:
Could you please also add a test that with a stat matcher that uses an `inclusion_list` ?

Reply:
Actually you'll also need tests with prefix matches with both inclusion and exclusion lists (to prevent regressions in the future).

Reply:
Done